examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,240 @@
+	}
+	if nc := os.Getenv("NumChats"); nc != "" {
+		num, err := strconv.Atoi(nc)
+		if err != nil {
+			log.Fatalf("Could not parse NumChats: %v", err)
+		}
+		numChats = &num
+	}
+
+	log.Print("Creating SDK instance")
+	s, err := sdk.NewSDK()
+	if err != nil {
+		log.Fatalf("Could not connect to sdk: %v", err)
+	}
+
+	log.Print("Starting Health Ping")
+	// TODO: Is "cancel" still being used even if no method explicitly uses it?
+	ctx, _ := signals.NewSigKillContext()
+	go doHealth(s, ctx)
+
+	log.Print("Marking this server as ready")
+	if err := s.Ready(); err != nil {
+		log.Fatalf("Could not send ready message")
+	}
+
+	// Start up TCP listener so the user can interact with the GenAI endpoint manually
+	if simConn == nil {
+		go tcpListener(port, s, compConn)
+	} else {
+		go handleChat(*prompt, compConn, simConn, *numChats)
+	}
+
+	<-sigCtx.Done()
+	os.Exit(0)
+}
+
+func initClient(endpoint string, context string) (*connection) {
+	// TODO: create option for a client certificate
+	client := &http.Client{}
+	conn := &connection{client: client, endpoint: endpoint, context: context}
+	return conn
+}
+
+type connection struct {
+	client   *http.Client
+	endpoint string // full base URL for API requests
+	context  string
+	// TODO: create options for routes off the base URL
+}
+
+type GenAIRequest struct {
+	Context         string  `json:"context,omitempty"`
+	Prompt          string  `json:"prompt"`
+}
+
+func handleGenAIRequest(prompt string, clientConn *connection) (string, error) {
+	jsonRequest := GenAIRequest{
+		Context: clientConn.context,
+		Prompt: prompt,
+	}
+	jsonStr, err := json.Marshal(jsonRequest)
+	if err != nil {
+		return "", fmt.Errorf("unable to marshal json request: %v", err)
+	}
+
+	req, err := http.NewRequest("POST", clientConn.endpoint, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return "unable create http POST request", err
+	}
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := clientConn.client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("unable to post request: %v", err)
+	}
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("unable to read response body: %v", err)
+	}
+	defer resp.Body.Close()
+	body := string(responseBody)
+
+	if resp.StatusCode != 200 {
+		err = fmt.Errorf("Status: %s, Body: %s", resp.Status, body)
+	}
+	return string(responseBody) + "\n", err
+}
+
+// Two AIs (connection endpoints) talking to each other
+func handleChat(prompt string, conn1 *connection, conn2 *connection, numChats int) {
+	if numChats <= 0 {
+		return
+	}
+	response, err := handleGenAIRequest(prompt, conn1)
+	if err != nil {
+		log.Fatalf("could not send request: %v", err)
+	} else {
+		log.Printf("%d PROMPT: %s\nRESPONSE: %s\n", numChats, prompt, response)
+	}
+
+	numChats -= 1
+	handleChat(response, conn2, conn1, numChats)
+}
+
+func tcpListener(port *string, s *sdk.SDK, clientConn *connection) {
```

Comment:
I don't think we need to pass the `sdk.SDK` down through the interactive part. It's done in the simple game server so that you can interactively control the game server's calls into the SDK. But here the SDK should just be in the background and not part of the main program flow.